### ISSUE TITLE
npm@5.2.0 installation make webpack can't resolve private scoped package

### ISSUE THREAD

**Issue by qiyuan-wang** [npm/npm#17741]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

I have a project with some private scope dependencies.

Its dependencies structure described as:
```
project
  |-@p/dep1
    |-@p/dep2
    |-@p/dep3
    |-@p/dep4 (package installed with problem)
  |-@p/dep2
  |-@p/dep3
```

And under npm@5.1.0, packages are installed like:
```
project
  |- node_modules
        |- @p
            |- dep1
            |- dep2
            |- dep3
            |- dep4
```
All packages are flatten installed, and I make webpack work with config:
```
{
  resolve: {
    modules: [path.resolve('./node_modules')],
    ... // others
  }
  ... // others
}
```

But under npm@5.2.0, they are installed as:
```
project
  |- node_modules
        |- @p
            |- dep2
            |- dep3
            |- dep1
                |- node_modules
                      |- @p
                          |- dep4
```
p/dep4 is nested in @p/dep1's node_modules, and webpack won't find it unless I change its config to:
```
{
  resolve: {
    modules: [
      path.resolve(__dirname, './node_modules'), 
      path.resolve(__dirname, './node_modules/@p/dep1/node_modules')
    ],
    ... // others
  }
  ... // others
}
```

Why p/dep4 is nested?

**Comment by jsg2021** [comment]
I've hit this too... but its not limited to private/scoped packages.

I have scripts package (inspired by react-scripts) that is an umbrella package that has my toolchain dependencies, It's expecting the dependencies to be flattened into the root node_modules dir where possible... I have a second scripts package that is an extension of my base scripts package... adds some dependencies and scripts, etc... and another that add additional deps (kind of like a maven pom) On npm 5.1, everything was happy and all the dependencies were hoisted to the root. In 5.2, deps are nested in a node_modules dir where they stop being explicitly depended on.

**Comment by jsg2021** [comment]
I suspect #17678 did this... and its intentional. :(

**Comment by jsg2021** [comment]
Oh it was reverted. #17717

**Comment by qiyuan-wang** [comment]
@jsg2021, thank your for the links. So, we will wait for the next release.

**Comment by qiyuan-wang** [comment]
closed for npm@5.3.0 release.
